### PR TITLE
Add augmentations

### PR DESCRIPTION
**Pull request by alicjak1519**
A couple of augmentations were added. :)

### CODE REVIEW

File: datasets/dataset_creator.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        train_len, val_len = self._get_split_lens(dataset)
+        train_subset, _ = torch.utils.data.random_split(dataset, [train_len, val_len])
+        return train_subset
+
+    def get_val_subset(self) -> torch.utils.data.dataset.Subset:
+        val_transforms = self._get_val_transforms()
+        dataset = self._get_video_frame_datasets(val_transforms)
+        train_len, val_len = self._get_split_lens(dataset)
+        _, val_subset = torch.utils.data.random_split(dataset, [train_len, val_len])
+        return val_subset
```
Comment (alicjak1519) [inline datasets/dataset_creator.py:35]: There is a need to apply basic transforms for the validation dataset and basic + fancy augmenting transforms for the training dataset. Instead of splitting this dataset and then modifying transforms, there is a double dataset generation. I'm not sure if it's ok but this solution was recommended [here](https://discuss.pytorch.org/t/changing-transforms-after-creating-a-dataset/64929/4)

File: train.py [hunk 2]
```diff
@@ -108,51 +107,16 @@ def main(args):
...
-        datasets.append(dataset)
-    dataset = torch.utils.data.ConcatDataset(datasets)
-    
-    # split into train/val
-    train_val_ratio = args.ratio
-    train_len = round(len(dataset) * train_val_ratio)
-    val_len = len(dataset) - train_len
-    train, val = random_split(dataset, [train_len, val_len])
+
+    dataset_creator = DatasetCreator(args.data, args.classification_mode, args.num_segments, args.time,
```
Comment (alicjak1519) [inline train.py:111]: I created this DatasetCreator class to simplify the train.py script and move the logic of creating the dataset to another file. What do you think about it?

File: datasets/dataset_creator.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        ]
+        tensor_augmentations = [
+            T.RandomErasing(),
+            T.RandomRotation(degrees=5),
+            T.ColorJitter(brightness=0.5,
+                          contrast=0.5,
+                          saturation=0.5,
+                          hue=0.2)
+        ]
```
Comment (alicjak1519) [inline datasets/dataset_creator.py:75]: This is a clue of this PR. ;) Just a couple of simple augmentations. Maybe can I add something else?

**Review by alicjak1519 — COMMENTED** [review]
I'd like to test the effectiveness of the version with and without augmentations and then we can decide about adding them.

File: datasets/dataset_creator.py (hunk 3 above)
Comment (AgaMiko) [reply datasets/dataset_creator.py:75]: Good! I would only change the color jitter params to 0.1 max for all params, maybe 0.05 for hue, as increasing brightness by 0.5 is very strong, and most images would end up white :D

Comment (alicjak1519) [reply datasets/dataset_creator.py]: Done ✅

**Comment by AgaMiko** [comment]
Looks good to me!

File: datasets/dataset_creator.py (hunk 1 above)
Comment (AgaMiko) [reply datasets/dataset_creator.py:35]: As long as the seed works well, I'm fine with it! We will probably remove random split so that's ok I think.

**Review by AgaMiko — APPROVED** [review]

File: datasets/dataset_creator.py (hunk 1 above)
Comment (majsylw) [reply datasets/dataset_creator.py]: @alicjak1519 have you checked if received train_subset in both function is the same? I know that in get_val_subset() you do not receive train_subset, but you can check if the training set and the validation set are separate - none repeted image ;-)
I have a little doubt about the seeding here

**Review by MilenaOlech — APPROVED** [review]

**Review by aga0 — APPROVED** [review]
Good job @alicjak1519! I went through all comments and I don't have any additional insights

**Comment by alicjak1519** [comment]
Screenshot 2022-03-12 at 15 27 47
It seems that applying augmentations improves loss a little bit so, although the time of learning is longer, I would like to merge this PR. :)

File: datasets/dataset_creator.py (hunk 1 above)
Comment (alicjak1519) [reply datasets/dataset_creator.py]: You are right - there are a couple of common indices. I'll fix it. :)

Comment (alicjak1519) [reply datasets/dataset_creator.py]: Fixed! :) I checked it by printing 
```print(list(set(train_subset.indices) & set(val_subset.indices)))```

and receiving an empty list. :)

Comment (majsylw) [reply datasets/dataset_creator.py]: good!

**Review by majsylw — APPROVED** [review]
Just before merging check please how to put parameters for augumentation into yaml file.
To have possibility to easly choose proper one for each validation and training ;-)

File: train.py [hunk 4]
```diff
@@ -127,54 +127,32 @@ def main(args):
...
+                                           bool(model_config["augmentations"]["apply_random_rotation"]),
+                                           bool(model_config["augmentations"]["apply_color_jitter"]),
+                                           int(model_config["augmentations"]["resize_size"]),
+                                           int(model_config["augmentations"]["center_crop_size"]),
+                                           float(model_config["augmentations"]["random_erasing_probability"]),
+                                           int(model_config["augmentations"]["random_rotation_degree"]),
+                                           float(model_config["augmentations"]["color_jitter_brightness"]),
+                                           float(model_config["augmentations"]["color_jitter_contrast"]),
+                                           float(model_config["augmentations"]["color_jitter_saturation"]),
+                                           float(model_config["augmentations"]["color_jitter_hue"]))
```
Comment (alicjak1519) [inline train.py:145]: This object enables to easily pass augmentations parameters from the config in case of the need. :) 
@majsylw is that what you meant? :)
